hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -1243,7 +1250,11 @@ public String getPreCombineField() {
   }
 
   public String getWritePayloadClass() {
-    return getString(WRITE_PAYLOAD_CLASS_NAME);
+    RecordPayloadType payloadType = RecordPayloadType.valueOf(getString(WRITE_PAYLOAD_TYPE));
+    if (payloadType.equals(RecordPayloadType.CUSTOM)) {
+      return getString(WRITE_PAYLOAD_CLASS_NAME);
```

Comment:
How about the user does not set up the payload type explicitly, is there any inference for that?

Reply:
Actually, this method is not used. I am going to remove it. However, to your point about inference, we do that in `HoodieTableConfig` or `HoodieWriterUtils`